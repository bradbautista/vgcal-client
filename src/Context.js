import React from 'react'

const Context = React.createContext({
    releases: [],    
})

export default Context