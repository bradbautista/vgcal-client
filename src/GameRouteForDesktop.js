import React, { Component } from 'react';
import Context from './Context';
import FullCalendar from '@fullcalendar/react';
import dayGridPlugin from '@fullcalendar/daygrid';
import './GameRoute.css';

export default class GameRouteForDesktop extends Component {
  
    constructor(props) {
        super(props);
        this.state = {
          timer: 0,
        };
    }

    static contextType = Context;

    waitForGame = () => {

        const gameName = this.props.match.params.game;

        const selectedGame = this.context.releases.filter(game => game.gameUrl.split('/').pop() === gameName).shift();

        if (typeof selectedGame !== "undefined") {
            this.context.setGame(selectedGame);
            this.context.setLoading(false);
            this.context.setOpen(true);
        } else if (this.state.timer > 5) {
            this.context.setError(true);
            this.context.setLoading(false);
        }
        else {
            this.setState({ timer: this.state.timer + 1 })
            this.context.setLoading(true);
            setTimeout(this.waitForGame, 1000);
        }
    }

    componentDidMount() {   
        this.waitForGame();
    }

    render() {

        return (
            <>               
                <FullCalendar 
                defaultView="dayGridMonth" 
                plugins={[ dayGridPlugin ]} 
                events={ this.context.releases }
                eventClick={ this.context.handleClick }
                ref={this.context.calendar}
                datesRender={ (info) => this.context.paintUrlDesktop(info) }
                height={ "auto" }
                />
            </>
        );
    }
}