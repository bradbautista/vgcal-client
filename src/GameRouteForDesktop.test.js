import React from 'react';
import ReactDOM from 'react-dom';
import GameRouteForDesktop from './GameRouteForDesktop';
import Context from './Context';
import { BrowserRouter } from 'react-router-dom';
import EnzymeToJson from 'enzyme-to-json';
import { mount } from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

// Need to use Enzyme for these route snapshot tests b/c of 
// react-test-renderer / js-dom issues

configure({ adapter: new Adapter() });

const someFunc = () => { console.log('Do stuff') };

const match = { params : { game: 'Namcot-Collection'} }

const games = [{"id":6990,"boxart_url":"https://giantbomb1.cbsistatic.com/uploads/original/16/164924/3177373-0778025400-mediu.png","game_name":"Namcot Collection","game_description":"A collection of NES titles from Namco, from under their Namcot banner.","platforms":"Nintendo Switch","release_date_utc":"June 18, 2020","release_date_iso":"2020-06-18","release_day":"18","release_month":"06","release_year":"2020","release_quarter":null}]

it('Renders the calendar without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <BrowserRouter>
        <Context.Provider value={ { games: games, handleClick: () => { someFunc() }, calendar: React.createRef(), paintUrlDesktop: () => { someFunc() }, setGame: () => { someFunc() }, setLoading: () => { someFunc() }, setError: () => { someFunc() }, setOpen: () => { someFunc() } } }>
            <GameRouteForDesktop match={match} />
        </Context.Provider>
    </BrowserRouter>, div);
  ReactDOM.unmountComponentAtNode(div);
});

test("resolves the url and renders the calendar", () => {

    const subject = mount(
        <BrowserRouter>
            <Context.Provider value={ { games: games, handleClick: () => { someFunc() }, calendar: React.createRef(), paintUrlDesktop: () => { someFunc() }, setGame: () => { someFunc() }, setLoading: () => { someFunc() }, setError: () => { someFunc() }, setOpen: () => { someFunc() } } }>
                <GameRouteForDesktop match={match} />
            </Context.Provider>
        </BrowserRouter>

    );
    expect(EnzymeToJson(subject)).toMatchSnapshot();
});
