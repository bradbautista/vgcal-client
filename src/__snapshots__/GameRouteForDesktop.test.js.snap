// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`resolves the url and renders the calendar 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <GameRouteForDesktop
      match={
        Object {
          "params": Object {
            "game": "Namcot-Collection",
          },
        }
      }
    >
      <FullCalendar
        datesRender={[Function]}
        defaultView="dayGridMonth"
        eventClick={[Function]}
        height="auto"
        plugins={
          Array [
            Object {
              "calendarInteractions": Array [],
              "cmdFormatter": undefined,
              "componentInteractions": Array [],
              "datePointTransforms": Array [],
              "dateSelectionTransformers": Array [],
              "dateSpanTransforms": Array [],
              "defaultView": "dayGridMonth",
              "deps": Array [],
              "elementDraggingImpl": undefined,
              "eventDefMutationAppliers": Array [],
              "eventDefParsers": Array [],
              "eventDragMutationMassagers": Array [],
              "eventDropTransformers": Array [],
              "eventResizeJoinTransforms": Array [],
              "eventSourceDefs": Array [],
              "externalDefTransforms": Array [],
              "id": "5",
              "isDraggableTransformers": Array [],
              "isPropsValid": null,
              "namedTimeZonedImpl": undefined,
              "optionChangeHandlers": Object {},
              "recurringTypes": Array [],
              "reducers": Array [],
              "themeClasses": Object {},
              "viewContainerModifiers": Array [],
              "viewPropsTransformers": Array [],
              "views": Object {
                "dayGrid": [Function],
                "dayGridDay": Object {
                  "duration": Object {
                    "days": 1,
                  },
                  "type": "dayGrid",
                },
                "dayGridMonth": Object {
                  "duration": Object {
                    "months": 1,
                  },
                  "fixedWeekCount": true,
                  "monthMode": true,
                  "type": "dayGrid",
                },
                "dayGridWeek": Object {
                  "duration": Object {
                    "weeks": 1,
                  },
                  "type": "dayGrid",
                },
              },
            },
          ]
        }
      >
        <div />
      </FullCalendar>
    </GameRouteForDesktop>
  </Router>
</BrowserRouter>
`;
